File: Utils/Environment.cs
using Utils.Enums;

namespace Utils
{
    public static class Environment
    {
        public static string Name => System.Environment.GetEnvironmentVariable("ENVIRONMENT") ?? "Development";

        public static EnvironmentEnum Type => Name switch
        {
            "Development" => EnvironmentEnum.Development,
            "Testing" => EnvironmentEnum.Testing,
            "Production" => EnvironmentEnum.Production,
            _ => throw new Exception($"Environment {Type} unknown.")
        };

        public static bool IsDevelopment => Type == EnvironmentEnum.Development;

        public static bool IsTesting => Type == EnvironmentEnum.Testing;

        public static bool IsProduction => Type == EnvironmentEnum.Production;

        public static string CosmosDBEndpoint => IsDevelopment ?
            Constants.COSMOS_DB_EMULATOR_ENDPOINT :
            System.Environment.GetEnvironmentVariable("COSMOS_DB_ENDPOINT") ??
            throw new Exception("Environment variable 'COSMOS_DB_ENDPOINT' missing.");

        public static string CosmosDBKey => IsDevelopment ?
            Constants.COSMOS_DB_EMULATOR_KEY :
            System.Environment.GetEnvironmentVariable("COSMOS_DB_KEY") ??
            throw new Exception("Environment variable 'COSMOS_DB_KEY' missing.");
    }
}

File: FreeWaterTips.ReactNative/src/utils/Environment.ts
export class Environment {
  static get name(): string {
    const environment = process.env.ENVIRONMENT || process.env.NODE_ENV;

    if (!environment) {
      throw new Error('ENVIRONMENT environment variable is not set.');
    }

    return environment;
  }

  static get isDevelopment(): boolean {
    return this.name !== 'Production' && this.name !== 'Testing';
  }

  static get isTesting(): boolean {
    return this.name === 'Testing';
  }

  static get isProduction(): boolean {
    return this.name === 'Production';
  }
}
